### PR TITLE
NodeJS Week1 from Rabia

### PR DESCRIPTION
**Pull request by Damla78**

### CODE REVIEW

File: week1/homework/src/server.js [hunk 1]
```diff
@@ -1,17 +1,40 @@
 'use strict';
 
+const url = require('url');
 const http = require('http');
+// const fs = require('fs');
```
Comment (keepitterron) [inline week1/homework/src/server.js]: when submitting work for code review, remember to always cleanup commented code :)

File: week1/homework/src/server.js [hunk 2]
```diff
@@ -1,17 +1,40 @@
...
 
 /* `createServer` MUST return an instance of `http.Server` otherwise the tests
  * will fail.
  */
 function createServer(port) {
   let state = 10;
+  let newState = state;
```
Comment (keepitterron) [inline week1/homework/src/server.js]: this is amazing Rabia, and you're the only one to do this so far.
it's very important to not mutate the state, we will see why in the next lessons :)

File: week1/homework/src/server.js [hunk 3]
```diff
@@ -1,17 +1,40 @@
...
 
   const server = http.createServer((request, response) => {
     // TODO: Write your homework code here
-  });
+    const requestedUrl = url.parse(request.url, true);
```
Comment (keepitterron) [inline week1/homework/src/server.js]: this doesn't hurt and it works fine, but you don't need to parse the url for this particular case,
`request.url` would work just fine :)

File: week1/homework/src/server.js [hunk 4]
```diff
@@ -1,17 +1,40 @@
...
+      response.writeHead(200, { 'Content-type': 'application/json' });
+      newState = newState + 1;
+      response.end(JSON.stringify({ state: newState }));
+    } else if (requestedUrl.pathname === '/subtract') {
+      response.writeHead(200, { 'Content-type': 'application/json' });
+      newState = newState - 1;
+      response.end(JSON.stringify({ state: newState }));
+    } else if (requestedUrl.pathname === '/reset') {
+      response.writeHead(200, { 'Content-type': 'application/json' });
+      newState = state;
```
Comment (keepitterron) [inline week1/homework/src/server.js]: 🎉

File: week1/homework/src/server.js [hunk 5]
```diff
@@ -1,17 +1,40 @@
...
     // TODO: Write your homework code here
-  });
+    const requestedUrl = url.parse(request.url, true);
 
+    if (requestedUrl.pathname === '/state') {
+      response.writeHead(200, { 'Content-type': 'application/json' });
+      response.end(JSON.stringify({ state: newState }));
+    } else if (requestedUrl.pathname === '/add') {
+      response.writeHead(200, { 'Content-type': 'application/json' });
+      newState = newState + 1;
```
Comment (keepitterron) [inline week1/homework/src/server.js]: maybe you already know and just prefer it this way, you can make this shorter by doing `newState++` to increment a variable and `newState--` to decrement :)

File: week1/homework/src/server.js (hunk 1 above)
Comment (Damla78) [reply week1/homework/src/server.js]: I cleaned.

File: week1/homework/src/server.js (hunk 5 above)
Comment (Damla78) [reply week1/homework/src/server.js]: I changed.

File: week1/homework/src/server.js [hunk 6]
```diff
@@ -1,20 +1,45 @@
...
+        break;
+      default:
+        writeRespond(404, { error: 'Not found' }, response);
+    }
   });
-
   return server;
 }
 
+function writeRespond(statuCode, stateMsg, response) {
```
Comment (keepitterron) [inline week1/homework/src/server.js]: great that you're passing the `response` to the function. This way this function is totally autonomous and it's easier to test (you'll learn about unit tests in one of the future lessons)

**Pull request closed without merging**